examples/js/lights/LightProbeGenerator.js

```diff
@@ -116,6 +116,105 @@ THREE.LightProbeGenerator = {
 
 		return new THREE.LightProbe( sh );
 
+	},
+
+	fromCubeCamera: function ( renderer, cubeCamera ) {
+		
+		// The cubeCamera renderTarget must be set to RGBA in order to make readRenderTargetPixels works
+		var norm, lengthSq, weight, totalWeight = 0;
+
+		var coord = new THREE.Vector3();
+
+		var dir = new THREE.Vector3();
+
+		var color = new THREE.Color();
+
+		var shBasis = [ 0, 0, 0, 0, 0, 0, 0, 0, 0 ];
+
+		var sh = new THREE.SphericalHarmonics3();
+		var shCoefficients = sh.coefficients;
+
+		for ( var faceIndex = 0; faceIndex < 6; faceIndex ++ ) {
+
+			var imageWidth = cubeCamera.renderTarget.width; // assumed to be square
+			var data = new Uint8Array(imageWidth * imageWidth * 4);
+			renderer.readRenderTargetPixels(cubeCamera.renderTarget, 0, 0, imageWidth, imageWidth, data, faceIndex);
+
+			var pixelSize = 2 / imageWidth;
+
+			for ( var i = 0, il = data.length; i < il; i += 4 ) { // RGBA assumed
+
+				// pixel color
+				color.setRGB( data[ i ] / 255, data[ i + 1 ] / 255, data[ i + 2 ] / 255 );
+				// pixel coordinate on unit cube
+
+				if (renderer.gammaOutput) {
```

Comment:
All light probes must be in linear space. It has nothing to do with render settings.

Reply:
Yes and this is why it depends on the renderer gammaOutput, keep in mind that the LightProbes is computed from a cubeCamera. During the rendering stage of the cubemap, gamma correction is applied if renderer.gammaOutput is true.

Reply:
The name is meaningful I think. I do not understand what is the matter with it
I have to read pixels back from the GPU to the CPU. The existing method is a little bit different and based on images. It seams a lot more easy for people who are not very confortable with CG to handle this seamlessly

Reply:
> Your environment map is a Cornell box with colors of the form `[ 0, 1, 1 ]`, so converting to and from sRGB space does nothing. Hence it is difficult to test if your color-space is correct.

True, I will check this tomorrow with another one

Reply:
>During the rendering stage of the cubemap, gamma correction is applied if renderer.gammaOutput is true.

Right. And I have suggested repeatedly that you handle the cube camera at the application level and pass in `cubeCamera.renderTarget` to a properly-named method.

I do appreciate you doing this, however. :-)

In fact, it should be implemented on the GPU. Then we can auto-generate irradiance maps from any WebGLRenderTargetCube.

Reply:
A fromCubeRenderTarget will be ok for you ?
Yes, it should be on the GPU too, there is another pull request for this https://github.com/mrdoob/three.js/pull/16152

Reply:
I feel pretty strongly that would be a better API. That is why I asked you if there was a reason why you didn't do it that way.

If you want, we can ask @mrdoob what he prefers. I don't want to waste your time.

Reply:
This is indeed a matter of point of view, maybe @mrdoob can give it conceptual advice ;-)